common/convert/config.go

```diff
@@ -0,0 +1,118 @@
+package convert
+
+import (
+	"encoding/json"
+	"strings"
+
+	dsc "github.com/aserto-dev/go-directory/aserto/directory/common/v3"
+	"github.com/aserto-dev/scim/common"
+	"github.com/aserto-dev/scim/common/config"
+	"github.com/pkg/errors"
+)
+
+type TemplateName int
+
+const (
+	Users TemplateName = iota
+	UsersGroups
+	UsersGroupsRoles
+)
+
+var ErrInvalidConfig = errors.New("invalid config")
+
+func (t TemplateName) String() string {
+	switch t {
+	case Users:
+		return "users"
+	case UsersGroups:
+		return "users-groups"
+	case UsersGroupsRoles:
+		return "users-groups-roles"
+	}
+
+	return "unknown"
+}
+
+type TransformConfig struct {
+	*config.Config
+	template           TemplateName
+	IdentityObjectType string `json:"identity_object_type,omitempty"`
+	IdentityRelation   string `json:"identity_relation,omitempty"`
+}
+
+func NewTransformConfig(cfg *config.Config) (*TransformConfig, error) {
+	template := Users
+
+	if cfg.Group != nil {
+		template = UsersGroups
+		if cfg.Role != nil {
```

Comment:
What happens if Group is nil but Role isn't?

Reply:
We don't have a template that supports roles but not groups... in any case, this is going away in the next PR